openapi/generator.go

```diff
@@ -1161,3 +1169,19 @@ func fieldNameFromTag(sf reflect.StructField, tagName string) string {
 	}
 	return name
 }
+
+/// parseExampleValue is used to transform the string representation of the example value to the correct type.
+func parseExampleValue(schemaType, value string) (interface{}, error) {
+	switch schemaType {
```

Comment:
While it may sound ok, the `schemaType` received here is a literal string value that emanate from the `schema.Type` field, which is intended for the OpenAPI spec, it doesn't represent a native Go type.
Instead, we should take a `reflect.Type`. In `newSchemaFromStructField`, you can pass the value of `sf.Type` which represents the reflection type of the struct field.